lang/c++/api/Decoder.hh

```diff
@@ -47,11 +47,11 @@ namespace avro {
  */
 class AVRO_DECL Decoder {
 public:
-    virtual ~Decoder() { };
+    virtual ~Decoder() = default;;
     /// All future decoding will come from is, which should be valid
     /// until replaced by another call to init() or this Decoder is
     /// destructed.
-    virtual void init(InputStream& is) = 0;
+    virtual void init(InputStream &is) = 0;
```

Comment:
based on all the reformatting, it seems like the more accepted version is to put the reference with the type?

Reply:
This change makes sense because 
        int& i, j;
is actually
        int &i, j

Reply:
This is a highly dogmatic topic.  If there isn't a style guide in place for Avro C++ that addresses this already, I think it is worth at least a heads up and separating it into a different PR.